lib/capistrano/tasks/capistrano2.rb

```diff
@@ -34,11 +34,7 @@
     def for_each_process(sidekiq_role, &block)
       sidekiq_processes = fetch(:"#{ sidekiq_role }_processes") rescue 1
       sidekiq_processes.times do |idx|
-        if idx.zero? && sidekiq_processes <= 1
-          pid_file = fetch(:sidekiq_pid)
-        else
-          pid_file = fetch(:sidekiq_pid).gsub(/\.pid$/, "-#{idx}.pid")
-        end
+        pid_file = fetch(:sidekiq_pid).gsub(/\.pid$/, "-#{idx}.pid")
```

Comment:
The problems all stem from the fact that capistrano-sidekiq is now incompatible with the command line tool ‘sidekiq’ with the -C option for a config file.

For a week we’ve been chasing unexpected multiple instances of sidekiq running against the same queues, unable to figure out where the errant processes have been coming from.  We use multiple tools to start and stop sidekiq, and we really, really need all of the tools we use to do the same thing, not least because we have a hard ceiling on sidekiq concurrency at 12, above which our solr server will stop accepting connections and bring down the whole app.

I do understand the benefit of the naming scheme for those running multiple sidekiq instances on a single machine, but for anyone else it is completely unexpected.  And in our case at least, it breaks system management, which breaks the app.

We’ve reverted to 0.5.3.


> On Dec 4, 2015, at 2:27 PM, Abdelkader Boudih <notifications@github.com> wrote:
> 
> In lib/capistrano/tasks/capistrano2.rb:
> 
> > @@ -34,11 +34,7 @@
> >      def for_each_process(sidekiq_role, &block)
> >        sidekiq_processes = fetch(:"#{ sidekiq_role }_processes") rescue 1
> >        sidekiq_processes.times do |idx|
> > -        if idx.zero? && sidekiq_processes <= 1
> > -          pid_file = fetch(:sidekiq_pid)
> > -        else
> > -          pid_file = fetch(:sidekiq_pid).gsub(/\.pid$/, "-#{idx}.pid")
> > -        end
> > +        pid_file = fetch(:sidekiq_pid).gsub(/\.pid$/, "-#{idx}.pid")
> 
> what serious problems ?
> 
> —
> Reply to this email directly or view it on GitHub.
>